rust/main/agents/scraper/src/chain_scraper/mod.rs

```diff
@@ -249,28 +269,24 @@ impl HyperlaneSqlDb {
                 .collect::<HashMap<_, _>>();
 
             for (block_ref, _) in blocks_to_insert.drain(..) {
-                block_ref.id = hashes[&block_ref.hash];
+                hashes.get(&block_ref.hash).map(|id| block_ref.id = *id);
             }
         }
 
-        // ensure we have updated all the block ids and that we have info for all of
-        // them.
-        #[cfg(debug_assertions)]
-        for (hash, block) in blocks.iter() {
-            let block = block.as_ref().unwrap();
-            assert_eq!(hash, &block.hash);
-            assert!(block.id > 0);
-        }
-
-        Ok(blocks
+        let ensured_blocks = blocks
             .into_iter()
-            .map(|(hash, block_info)| block_info.unwrap()))
+            .filter_map(|(hash, block_info)| block_info.map(|info| info))
```

Comment:
could we write as 
```suggestion
            .filter_map(|(hash, block_info)| block_info)
```

Reply:
Would also be ok with merging the `filter` logic below with the `filter_map` logic here, to avoid iterating twice

Reply:
I suppose that `filter_map` and `filter` will be executed lazily when elements will be collected into a collection. So, as far as I understand, there will be only one pass but applying all the operations to each element.

I think you are proposing this:
```
            .filter_map(|(hash, block_info)| block_info.filter(|b| b.id != -1));
```

Reply:
Implemented.